src/agent/CMakeLists.txt

```diff
@@ -64,6 +64,14 @@ endif()
 set(OTBR_AGENT_USER "root" CACHE STRING "set the username running otbr-agent service")
 set(OTBR_AGENT_GROUP "root" CACHE STRING "set the group using otbr-agent client")
 
+if(OTBR_MDNS STREQUAL "mDNSResponder")
+    set(EXEC_START_PRE "ExecStartPre=service mdns start\n")
```

Comment:
what will happen if `EXEC_START_PRE` command fails?

Reply:
Then it prevents otbr-agent from starting. 
```
Job for otbr-agent.service failed because the control process exited with error code.
See "systemctl status otbr-agent.service" and "journalctl -xe" for details.
```

Reply:
SGTM.